Add components and asides guidance to `TRANSLATING.md` (#1466)
withastro/docs: TRANSLATING.md

// Rest of the file's content is here...
```

### Asides

> Most of our pages include stylish tip/note/caution blocks called "asides". As they're repeatedly used throughout the docs, we use a custom syntax to author them (and later replace with components during build), as you can see in the example below:

Reviewer: ```suggestion
Most of our pages include stylish tip/note/caution blocks called "asides". We use a custom syntax to author them which includes the type of aside (all lowercase) and optionally a custom title in square brackets. Here is an example of a "tip" found in the docs:
```

```
:::tip[Online previews]
Prefer to try Astro in your browser? Visit [astro.new](https://astro.new/) to browse our starter templates and spin up a new Astro project without ever leaving your browser.
:::
```

> Some asides contain custom inline labels inside `[square brackets]`, like the example above. You're free to translate these, as well as the text inside the aside block. Although, you **must not translate** the aside's type (like `:::tip`), as these are translated in the language's respective `i18n/nav.ts` file.

Reviewer: ```suggestion
**Do translate**: the custom inline labels inside `[square brackets]`, and the text inside the aside block. 

**Do not translate**: the aside's type (e.g. `:::tip`). These type names are instead translated once in each language's `i18n/nav.ts` file and are automatically replaced in your translated page as necessary.
```

Here is the above example correctly translated:

```
:::tip[オンラインプレビュー]
ブラウザでAstroを試してみませんか？[astro.new](https://astro.new/)では、スターターテンプレートを利用し、ブラウザから離れることなく、新しいAstroプロジェクトを立ち上げられます。
:::
```

### Components

> Astro allows us to include custom components in our Markdown pages, so we take full advantage of this to improve our documentation's overall quality. Take this fragment of the `islands.md` page, which renders a diagram, as an example:

Reviewer: ```suggestion
Astro allows us to import and include custom components in our Markdown pages. Take this fragment of the `islands.md` page, which renders a diagram, as an example:
```

```jsx
<IslandsDiagram>
    <Fragment slot="headerApp">Header (interactive island)</Fragment>
    <Fragment slot="sidebarApp">Sidebar (static HTML)</Fragment>
    <Fragment slot="main">
        Static content like text, images, etc.
    </Fragment>
    <Fragment slot="carouselApp">Image carousel (interactive island)</Fragment>
    <Fragment slot="footer">Footer (static HTML)</Fragment>
    <Fragment slot="source">Source: [Islands Architecture: Jason Miller](https://jasonformat.com/islands-architecture/)</Fragment>
</IslandsDiagram>
```

> As a rule of thumb, component names, imports, and slot names (like `slot="headerApp"`) **must not be translated nor modified in any way**. Although, you're completely free to translate slotted content as you see fit.

Reviewer: ```suggestion
**Do translate**: slotted content (content between opening and closing tags).

**Do not translate**: import statements in the frontmatter, component names, and slot names (like `slot="headerApp"`) 
```

// Rest of the file's content is here...
```

### Components

Astro allows us to include custom components in our Markdown pages, so we take full advantage of this to improve our documentation's overall quality. Take this fragment of the `islands.md` page, which renders a diagram, as an example:

```jsx
<IslandsDiagram>
    <Fragment slot="headerApp">Header (interactive island)</Fragment>
    <Fragment slot="sidebarApp">Sidebar (static HTML)</Fragment>
    <Fragment slot="main">
        Static content like text, images, etc.
    </Fragment>
    <Fragment slot="carouselApp">Image carousel (interactive island)</Fragment>
    <Fragment slot="footer">Footer (static HTML)</Fragment>
    <Fragment slot="source">Source: [Islands Architecture: Jason Miller](https://jasonformat.com/islands-architecture/)</Fragment>
</IslandsDiagram>

> 

Reviewer 2: Nit, but I guess no empty line here
```suggestion
```

```

As a rule of thumb, component names, imports, and slot names (like `slot="headerApp"`) **must not be translated nor modified in any way**. Although, you're completely free to translate slotted content as you see fit.

Here is the above example correctly translated:

```jsx
<IslandsDiagram>
    <Fragment slot="headerApp">Cabeçalho (ilha interativa)</Fragment>
    <Fragment slot="sidebarApp">Barra lateral (HTML estático)</Fragment>
    <Fragment slot="main">
        Conteúdo estático como texto, imagens, etc.
    </Fragment>
    <Fragment slot="carouselApp">Carrossel de imagens (ilha interativa)</Fragment>
    <Fragment slot="footer">Rodapé (HTML estático)</Fragment>
    <Fragment slot="source">Fonte: [Arquitetura em Ilhas: Jason Miller](https://jasonformat.com/islands-architecture/)</Fragment>
</IslandsDiagram>
```

Note that some of our components' labels are instead translated inside the language's respective `i18n/` files, as we explain in the [Translation Structure section](#2-ui-text).

> ### Asides

Reviewer 2: This is not critical but I wonder if the **Asides** section should come _above_ the **Components** section as it’s more likely to be relevant. Happy to leave it as is though if you think things flow better this way.

Author: That's a fair point! It does make more sense considering we are using it way more than components without custom syntax.

Most of our pages include stylish tip/note/caution blocks called "asides". As they're repeatedly used throughout the docs, we use a custom syntax to author them (and later replace with components during build), as you can see in the example below:

```
:::tip[Online previews]
Prefer to try Astro in your browser? Visit [astro.new](https://astro.new/) to browse our starter templates and spin up a new Astro project without ever leaving your browser.
:::
```

> Some asides contain custom inline labels inside `[square brackets]`, like the example above. You're free to translate these, as well as the text inside the backticks. Although, you **must not translate** the aside's type (like `:::tip`), as these are translated in the language's respective `i18n/nav.ts` file.

Reviewer 2: Do you mean inside the `:::`-delimited block rather than backticks?

```suggestion
Some asides contain custom inline labels inside `[square brackets]`, like the example above. You're free to translate these, as well as the text inside the aside block. Although, you **must not translate** the aside's type (like `:::tip`), as these are translated in the language's respective `i18n/nav.ts` file.
```

Author: Exactly, looks like my mind was thinking about code blocks instead of aside blocks 😅
